Update to fetch CUB via Thrust to prevent version mismatch

jrhemstad opened the pull request:
Fetching newer versions of CUB can cause problems because it will be picked up by Thrust which will complain about version mismatch. 

To avoid this, fetch CUB through Thrust and link both against `cuco` to avoid version mismatch issues.

jrhemstad committed:
Remove fetching CUB and just rely on it from CTK.

jrhemstad committed:
Merge remote-tracking branch 'origin/dev' into remove-cub-fetch

alliepiper commented:
If you want to keep tracking `main`, you can just fetch Thrust as long as it's a recursive clone. That will pull in CUB, no need to explicitly pull both.

Thrust should work with CPM already as it's supported the `add_subdirectory` usecase for a while now.

jrhemstad commented:
> If you want to keep tracking `main`, you can just fetch Thrust as long as it's a recursive clone.

I realized that cuco doesn't depend on anything from CUB `main` ToT. If/when we want to update that I'll change it to fetching Thrust recursively so it automatically gets the correct version of CUB.

In the meantime, the only thing we rely on is `BlockReduce`, so I think we should be fine just relying on whatever is in the CTK?

jrhemstad committed:
Update to fetch Thrust and CUB.

Instead of fetching CUB directly, fetch Thrust, which indirectly provides CUB and ensures they are compatible versions.

To prevent version mismatch, link both Thrust and CUB against cuco.
